tests/std/tests/P1222R4_flat_set/test.cpp

```diff
@@ -408,14 +395,73 @@ void test_non_static_comparer() {
     assert_all_requirements_and_equals(a, {9, 7, 5, -1});
 }
 
+template <template <class...> class Set>
+void test_extract_1() {
+    // Test that the container will be emptied, even if the container's move ctor exits via an exception.
 
-template <class C>
-void test_extract() {
-    constexpr int elements[]{1, 2, 3, 4};
-    C fs{1, 2, 3, 4};
-    auto cont = std::move(fs).extract();
-    assert(fs.empty());
-    assert(ranges::equal(cont, elements));
+    static bool will_throw = false;
+
+    class test_exception : public vector<int> {
+        using base = vector<int>;

```

Comment:
This should explicitly say `private:` before `base`, to follow our usual conventions. Occurs below for `class always_copy`.